feature-libs/product-configurator/rulebased/components/product-title/configurator-product-title.component.ts

```diff
@@ -44,14 +44,9 @@ export class ConfiguratorProductTitleComponent {
         .pipe(map((configuration) => ({ routerData, configuration })))
         .pipe(
           map((container) => {
-            switch (container.configuration.owner.type) {
-              case CommonConfigurator.OwnerType.PRODUCT:
-                return container.configuration.owner.id;
-              case CommonConfigurator.OwnerType.CART_ENTRY:
-                return container.routerData.productCode;
-              case CommonConfigurator.OwnerType.ORDER_ENTRY:
-                return container.configuration.overview?.productCode;
-            }
+            return !!container.routerData.productCode
+              ? container.routerData.productCode
+              : container.configuration.productCode;
```

Comment:
Please test order history case with product variant that retains its code

Reply:
I retested.
Actually both `configuration.productCode` and `overview.productCode` are defined for the untouched variant.